program/src/state.rs

```diff
@@ -111,12 +117,17 @@ pub struct Config {
     ///
     /// Stored as a `u64`, which includes a scaling factor of `1e9` to
     /// represent the threshold with 9 decimal places of precision.
+    // Jon: Just a nit, but if it goes out to 9 decimal places, you can use a u32 to
+    // save space, since the max is 1_000_000_000, right?
+    // On the flip side, we could simplify this if we just set it to a required raw
+    // number of votes. What do you think?
```

Comment:
Thinking about this some more, I don't think it's the right call -- Uri asked for percentage, so let's stick with that, even though it can change on every call to vote. We could potentially store the total staked amount during voting start

Reply:
I was wondering what you meant by "raw number of votes", since votes all have different stake weight. I like the percentage method a bit more. Even though the math is funky it's still quite trivial. I can change to `u32` as well, as suggested.

Reply:
Sorry, I meant the raw stake amount, rather than a proportion of total stake. So if there are 500 staked tokens, we can specify 250 as the required amount for a yes to pass. But as tokens are staked or destaked, that'll make it easier / harder to pass proposals